functions/api/helpers.js

```diff
@@ -100,3 +107,94 @@ export async function refreshKPILatestValue(kpiRef) {
     });
   }
 }
+
+/**
+ * Build and return KPI response object.
+ *
+ * `kpiSnapshot` is the Firestore KPI document snapshot.
+ */
+export async function buildKpiResponse(kpiSnapshot) {
+  const {
+    created,
+    createdBy,
+    currentValue,
+    edited,
+    editedBy,
+    name,
+    type,
+    updateFrequency,
+  } = kpiSnapshot.data();
+
+  const latestMeasurement = await kpiSnapshot.ref
+    .collection('progress')
+    .orderBy('timestamp', 'desc')
+    .limit(1)
+    .get()
+    .then((snapshot) => {
+      if (!snapshot.docs[0]) return null;
+      const { value, timestamp } = snapshot.docs[0].data();
+      return { value, timestamp: timestamp.toDate() };
+    });
+
+  const isStale = isKPIStale(updateFrequency, latestMeasurement);
```

Comment:
Kunne kanskje bare inlines nedenfor, for å matche slik det er gjort med de andre feltene?